### ISSUE TITLE
Heredocs are parsed incorrectly

### ISSUE THREAD

**Issue by dualbus** [koalaman/shellcheck#108]
```
dualbus@debian:~$ cat heredoc.bash 
unquoted=unquoted
cat <<eof0; echo $unquoted; cat <<eof1
0
eof0
1
eof1
dualbus@debian:~$ bash heredoc.bash 
0
unquoted
1
dualbus@debian:~$ shellcheck heredoc.bash 
dualbus@debian:~$
```

Shellcheck should complain about the unquoted expansion, but it doesn't. I assume that it's because it incorrectly assumes that nothing can follow the heredoc starting token, but according to POSIX the heredoc starts after the end of line, so it should parse it like this:

```
<<token; here goes more code, heredoc has not  started\n
heredoc starts here (after the \n)\n
token\n
```

and not like this:
```
<<token; shellcheck ignores this\n
heredoc starts here (after the \n)\n
token\n
```

**Comment by knirch** [comment]
Added this to https://github.com/koalaman/shellcheck/issues/563#issuecomment-181260712, probably a better fit here; 

this seems to be a fairly common pattern where I'd expect SC2086 to trigger
```bash
file="foo bar"
cat <<EOF >> $file
foobar
EOF
```

**Comment by koalaman** [comment]
Fixed in 2f0ae44
